ui/common/Input.tsx

```diff
@@ -1,39 +1,93 @@
-'use client'
-
-import RICIBs from 'react-individual-character-input-boxes'
-
-export default function Input({
-  amount,
-  onChange,
-}: {
-  amount: number
-  onChange: Function
-}) {
-  const handleChange = (value: string) => {
-    onChange(value)
-  }
-
-  return (
-    <RICIBs
-      amount={amount}
-      autoFocus
-      handleOutputString={handleChange}
-      inputProps={{
-        className: 'bg-transparent',
-        placeholder: '_',
-        style: {
-          fontSize: '20px',
-          width: '20px',
-          height: '20px',
-          margin: '0px',
-          borderRadius: '0px',
-          textAlign: 'center',
-          justifyContent: 'space-evenly',
-          outline: 'none',
-          fontFamily: 'var(--space-mono-font)',
-        },
-      }}
-      inputRegExp={/^[a-zA-Z0-9_.-]*$/}
-    />
-  )
-}
+'use client'
+
+import { useState } from 'react'
+
+interface UserInputProps {
+  onChange: Function
+  answer: string
+  hints?: boolean
+}
+
+export default function Input({ onChange, answer, hints }: UserInputProps) {
+  const [textAreaValue, setTextAreaValue] = useState('')
+  const [correctAns, setCorrectAns] = useState(false)
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (event.key === ' ') {
+      event.preventDefault()
+    }
+  }
+
+  const blankSpace = () => {
+    let underscores = '_'.repeat(answer.length)
+    underscores = underscores.substring(0, answer.length - textAreaValue.length)
+    return (
+      <>
+        {Array.from({ length: answer.length }, (_, i) => {
+          if (i >= answer.length - underscores.length)
+            return (
+              <span className="underscore" key={i}>
+                _
+              </span>
+            )
+          if (textAreaValue[i] === answer[i]) {
+            return (
+              <span className="overlay correct" key={i}>
+                {textAreaValue[i]}
+              </span>
+            )
+          } else if (textAreaValue[i] !== answer[i] && !!hints) {
+            return (
+              <span className="overlay incorrect" key={i}>
+                {textAreaValue[i]}
+              </span>
+            )
+          } else {
+            return (
+              <span className="overlay" key={i}>
+                {textAreaValue[i]}
+              </span>
+            )
+          }
+        })}
+      </>
+    )
+  }
+
+  return (
+    <form className="relative">
+      <textarea
+        onKeyDown={handleKeyDown}
+        onChange={(e) => {
+          let noSpace = e.target.value.replace(/[\s\u00A0]+/g, '').replace(/\r?\n|\r/g, '').slice(0, answer.length).toLowerCase()

```

Comment:
I'm not sure what this code is doing:
```js
e.target.value.replace(/[\s\u00A0]+/g, '').replace(/\r?\n|\r/g, '').slice(0, answer.length).toLowerCase()
```

Its use seems to be to remove linebreaks from the users input, if that is the case I think a better approach is to use the `keypress` event and check what keyCode is being pressed. If the keyCode matches certain characters you wish to block you can `return false`. This way you dont have to modify the value after the fact.

Reply:
This was intended to handle formatting when pasting in text, it is intended to catch linebreaks and whitespace characters, i refactored it to make it more clear what does what

Reply:
Okay I see, let's use [the paste event](https://developer.mozilla.org/en-US/docs/Web/API/Element/paste_event) for that though. Something like this:

```js
function handlePaste(event) {
  event.preventDefault()

  const pasteData = (event.clipboardData || window.clipboardData).getData('text')
    .replace(/\s+/g, '')

  document.execCommand('insertText', false, pasteData)
}
```

```jsx
<textarea
  onPaste={handlePaste}
  ...
/>
```

Reply:
wow well that does it! thanks for showing me this solution!

Reply:
I tried implementing this paste event but it no longer set the textarea to the clipboard remaining blank after pasting

Reply:
Hey @benalleng, are you referring to this code or have you not pushed your attempt at this to GitHub?

```jsx
  const handlePaste = (event) => {
    const formattedText = event
      .replace(/[\s\u00A0]+/g, '')
      .replace(/\r?\n|\r/g, '')
      .slice(0, answer.length)
      .toLowerCase()
    return formattedText
  }
  
  const handleChange = (event) => {
    let formattedText = handlePaste(event.target.value)
    ...
  })
  ```
  
  ```jsx
  <textarea
    onChange={handleChange}
    ...
  />
  ```

Reply:
I resolved this function with a slight adjustment to the onPaste handler